clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -160,17 +160,30 @@ private void reregister(KafkaMbean mbean) {
         }
     }
 
-    private static class KafkaMbean implements DynamicMBean {
+    public int mbeansQty() {
+        return this.mbeans.size();
+    }
+
+    protected static class KafkaMbean implements DynamicMBean {
+
         private final ObjectName objectName;
         private final Map<String, KafkaMetric> metrics;
 
-        public KafkaMbean(String mbeanName) throws MalformedObjectNameException {
-            this.metrics = new HashMap<String, KafkaMetric>();
-            this.objectName = new ObjectName(mbeanName);
+        public static KafkaMbean of(ObjectName objectName) {
+            try {
+                return new KafkaMbean(objectName);
+            } catch (MalformedObjectNameException e) {
+                throw new KafkaException("Error creating mbean with " + objectName, e);
+            }
+        }
+
+        private KafkaMbean(ObjectName objectName) throws MalformedObjectNameException {
```

Comment:
This exception is no longer possible since the constructor is taking `ObjectName`.